### PR TITLE
[RISCV][TTI] Add instruction cost for vp.select.

### PR DESCRIPTION
**Pull request by ElvisWang123**
This patch make instruction cost for vp.select  the same as its non-vp counterpart.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp [hunk 1]
```diff
@@ -1065,6 +1065,13 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
         *FOp, ICA.getArgTypes()[0], UI->getPointerAlignment(),
         UI->getOperand(1)->getType()->getPointerAddressSpace(), CostKind);
   }
+  case Intrinsic::vp_select: {
+    Intrinsic::ID IID = ICA.getID();
+    std::optional<unsigned> FOp = VPIntrinsic::getFunctionalOpcodeForVP(IID);
+    assert(FOp.has_value());
```
Comment (tschuett) [inline llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1071]: It seems to be a common pattern, but you cannot assert that an optional has a value.
```
if (FOp) {
 return getCmpSelInstrCost(*FOp, ICA.getReturnType(), ICA.getArgTypes()[0],
                              CmpInst::BAD_ICMP_PREDICATE, CostKind);
} {
 // I don't know
}
```

Comment (ElvisWang123) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1071]: The `VPIntrinsic::getFunctionalOpcodeForVP()` will return an optional with value when the VPIntrinsic has corresponding instruction opcode.
We can find the mapping table in the `llvm/include/llvm/IR/VPIntrinsics.def` which contains a macro called `VP_PROPERTY_FUNCTIONAL_OPC`. 
Any query of the opcode for VPIntrinsics with `VP_PROPERTY_FUNCTIONAL_OPC` will not return  `std::nullopt`.  [Source](https://llvm.org/doxygen/IntrinsicInst_8cpp_source.html#l00519)


In this case, the vp_select contains the `VP_PROPERTY_FUNCTIONAL_OPC(Select)` attribute so we can assert that the `FOp` has value. [Source](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/IR/VPIntrinsics.def#L720C1-L720C27)

Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp:1071]: I think you're right that getFunctionalOpcodeForVP is doing two things, checking if a VP intrinsic has a non VP equivalent and then also returning the opcode for said non VP equivalent 

We should probably refactor it in a separate PR.

File: llvm/test/Analysis/CostModel/RISCV/rvv-select.ll [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ; NOTE: Assertions have been autogenerated by utils/update_analyze_test_checks.py
-; RUN: opt -passes="print<cost-model>" 2>&1 -disable-output -mtriple=riscv64 -mattr=+v,+f,+d,+zfh,+zvfh -riscv-v-vector-bits-min=128 -riscv-v-fixed-length-vector-lmul-max=1 < %s | FileCheck %s
+; RUN: opt -passes="print<cost-model>" 2>&1 -disable-output -mtriple=riscv64 -mattr=+v,+f,+d,+zfh,+zvfh -riscv-v-vector-bits-min=128 -riscv-v-fixed-length-vector-lmul-max=1 < %s | FileCheck --check-prefixes=CHECK %s
+; RUN: opt -passes="print<cost-model>" 2>&1 -disable-output -mtriple=riscv64 -mattr=+v,+f,+d,+zfh,+zvfh -riscv-v-vector-bits-min=128 -riscv-v-fixed-length-vector-lmul-max=1 --type-based-intrinsic-cost=true < %s | FileCheck --check-prefixes=CHECK %s
```
Comment (lukel97) [inline llvm/test/Analysis/CostModel/RISCV/rvv-select.ll:3]: Nit I think we can remove the -check-prefix if the check lines are all the same

Comment (ElvisWang123) [reply llvm/test/Analysis/CostModel/RISCV/rvv-select.ll]: Removed, thanks!

**Review by lukel97 — APPROVED** [review]

**Pull request merged**